tofhir-server/src/main/scala/io/tofhir/server/endpoint/ToFhirServerEndpoint.scala

```diff
@@ -22,13 +23,16 @@ class ToFhirServerEndpoint(toFhirEngineConfig: ToFhirEngineConfig, webServerConf
 
   val terminologyServiceManagerEndpoint = new TerminologyServiceManagerEndpoint(toFhirEngineConfig)
 
-  val projectRepository: IProjectRepository = new ProjectFolderRepository(toFhirEngineConfig) // creating the repository instance globally as weed a singleton instance
-  val schemaRepository: ISchemaRepository = new SchemaFolderRepository(toFhirEngineConfig.schemaRepositoryFolderPath, projectRepository.asInstanceOf[ProjectFolderRepository])
-  val projectEndpoint = new ProjectEndpoint(toFhirEngineConfig, schemaRepository, projectRepository)
-  val fhirDefinitionsEndpoint = new FhirDefinitionsEndpoint(fhirDefinitionsConfig)
+  val projectRepository: ProjectFolderRepository = new ProjectFolderRepository(toFhirEngineConfig) // creating the repository instance globally as weed a singleton instance
+  val mappingRepository: MappingFolderRepository = new MappingFolderRepository(toFhirEngineConfig.mappingRepositoryFolderPath, projectRepository)
+  val schemaRepository: SchemaFolderRepository = new SchemaFolderRepository(toFhirEngineConfig.schemaRepositoryFolderPath, projectRepository)
+  val mappingJobRepository: JobFolderRepository = new JobFolderRepository(toFhirEngineConfig.jobRepositoryFolderPath, projectRepository)
+
+  // Initialize the projects by reading the resources available in the file system
+  projectRepository.initProjects(schemaRepository, mappingRepository, mappingJobRepository)
```

Comment:
I suggest to move db initialization logic into a separate class so that ProjectRepository does not depend on the others. Here is my suggestion: 

```suggestion
  val dbInitializer: IDBInitializer = new FolderDBInitializer(projectRepository, mappingRepository, schemaRepository, mappingJobRepository).init()
```
Then, we can move **ProjectFolderRepository.initProjects** implementation into **init** method.